src/dfx/src/commands/ledger/create_canister.rs

```diff
@@ -70,6 +70,7 @@ pub async fn exec(env: &dyn Environment, opts: CreateCanisterOpts) -> DfxResult
     })?;
 
     let agent = env.get_agent();
+    let calling_identity = agent.get_principal().map_err(|err| anyhow!(err))?;
```

Comment:
I ran into this just yesterday.  Should be ok with just this

```suggestion
    let calling_identity = agent.get_principal()?;
```